File: objectModel/TypeScript/__test__/Storage/ResourceAdapter.test.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import { ResourceAdapter } from '../../Storage';

describe('Cdm.Storage.ResourceAdapter', () => {
    /**
     * Tests if the calls to CreateCorpusPath return the expected corpus path.
     */
    it('TestCreateCorpusPath', () => {
        const adapter: ResourceAdapter = new ResourceAdapter();

        let path: string = adapter.createCorpusPath('Microsoft.CommonDataModel.ObjectModel.Resources/extensions/pbi.extension.cdm.json');
        expect(path)
            .toBe('/extensions/pbi.extension.cdm.json');

        path = adapter.createCorpusPath('Microsoft.CommonDataModel.ObjectModel.Resources/primitives.cdm.json');
        expect(path)
            .toBe('/primitives.cdm.json');
    });

    /**
     * Tests if the calls to CreateAdapterPath return the expected adapter path.
     */
    it('TestCreateAdapterPath', () => {
        const adapter: ResourceAdapter = new ResourceAdapter();

        let path: string = adapter.createAdapterPath('/extensions/pbi.extension.cdm.json');
        expect(path)
            .toBe('Microsoft.CommonDataModel.ObjectModel.Resources/extensions/pbi.extension.cdm.json');

        path = adapter.createAdapterPath('/primitives.cdm.json');
        expect(path)
            .toBe('Microsoft.CommonDataModel.ObjectModel.Resources/primitives.cdm.json');
    });

    /**
     * Tests if the files from the resource adapter can be read correctly.
     */
    it('TestReadAsync', async (done) => {
        const adapter = new ResourceAdapter();

        expect(await adapter.readAsync('/extensions/pbi.extension.cdm.json'))
            .not
            .toBeNull();
        expect(await adapter.readAsync('/primitives.cdm.json'))
            .not
            .toBeNull();

        done();
    });
});

File: objectModel/Java/objectmodel/src/test/java/com/microsoft/commondatamodel/objectmodel/storage/ResourceAdapterTest.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.storage;

import com.microsoft.commondatamodel.objectmodel.storage.ResourceAdapter;
import org.testng.Assert;
import org.testng.annotations.Test;

public class ResourceAdapterTest {
  private static final String ROOT = "Microsoft.CommonDataModel.ObjectModel.Resources";

  /**
   * Tests if the calls to CreateCorpusPath return the expected corpus path.
   */
  @Test
  public void testCreateCorpusPath() {
    final ResourceAdapter adapter = new ResourceAdapter();

    String path = adapter.createCorpusPath(ROOT + "/extensions/pbi.extension.cdm.json");
    Assert.assertEquals("/extensions/pbi.extension.cdm.json", path);

    path = adapter.createCorpusPath(ROOT + "/primitives.cdm.json");
    Assert.assertEquals("/primitives.cdm.json", path);
  }

  /**
   * Tests if the calls to CreateAdapterPath return the expected adapter path.
   */
  @Test
  public void testCreateAdapterPath() {
    final ResourceAdapter adapter = new ResourceAdapter();

    String path = adapter.createAdapterPath("/extensions/pbi.extension.cdm.json");
    Assert.assertEquals(ROOT + "/extensions/pbi.extension.cdm.json", path);

    path = adapter.createAdapterPath("/primitives.cdm.json");
    Assert.assertEquals(ROOT + "/primitives.cdm.json", path);
  }

  /**
   * Tests if the files from the resource adapter can be read correclty.
   */
  @Test
  public void testReadAsync() {
    final ResourceAdapter adapter = new ResourceAdapter();

    Assert.assertNotNull(adapter.readAsync("/extensions/pbi.extension.cdm.json").join());
    Assert.assertNotNull(adapter.readAsync("/primitives.cdm.json").join());
  }
}
